homeassistant/components/light/knx.py

```diff
@@ -111,26 +169,47 @@ def should_poll(self):
     @property
     def brightness(self):
         """Return the brightness of this light between 0..255."""
-        return self.device.current_brightness \
-            if self.device.supports_brightness else \
-            None
+        if self.device.supports_color:
+            if self.device.current_color is None:
+                return None
+            return max(self.device.current_color)
+        if self.device.supports_brightness:
+            return self.device.current_brightness
+        return None
 
     @property
     def hs_color(self):
         """Return the HS color value."""
         if self.device.supports_color:
-            return color_util.color_RGB_to_hs(*self.device.current_color)
+            rgb = self.device.current_color
+            if rgb is None:
+                return None
+            return color_util.color_RGB_to_hs(*rgb)
         return None
 
     @property
     def color_temp(self):
-        """Return the CT color temperature."""
+        """Return the color temperature in mireds."""
+        if self.device.supports_color_temperature:
+            kelvin = self.device.current_color_temperature
+            return color_util.color_temperature_kelvin_to_mired(kelvin) \
+                if kelvin is not None else DEFAULT_COLOR_TEMPERATURE
+        if self.device.supports_tunable_white:
+            relative_ct = self.device.current_tunable_white
+            return self.min_mireds + (((255 - relative_ct) / 255) *
+                                      (self.max_mireds - self.min_mireds)) \
```

Comment:
As Mireds is the go-to unit for HA it seemed unnecessary to calculate this in a foreign unit. Especially since all Mireds values in this calculation are constants.

Reply:
It matters. For your default ranges:

* 2700-6000 Kelvin, 50% is 4350 Kelvin
* 370-166 mireds, 50% is 268 mireds

However, 4350 Kelvin is 230 mireds.

Reply:
The slider is indeed not intuitive because it is in mireds while we usually think in Kelvin.

Reply:
So shall we leave it as it is for lights that represent their color temperature in percent (outside of the HA world)? 
Or shall we go with the correct values but lesser intuitive sliders?

Reply:
Do it right or you will get error reports that Home Assistant sets the wrong temperature (I have been there, several times).

Fixing the slider is an unrelated issue that could improve the UI for all lights.